src/component.py

```diff
@@ -198,18 +221,19 @@ def _get_table_header(self, t: TableDefinition):
         return header
 
     def _has_header_in_file(self, t: TableDefinition):
-        is_input_mapping_manifest = 'uri' in t._raw_manifest
-        has_header = True
+        is_input_mapping_manifest = t.stage == 'in'
         if t.is_sliced:
             has_header = False
-        elif t.columns and not is_input_mapping_manifest:
+        elif t.column_names and not is_input_mapping_manifest:
             has_header = False
         else:
             has_header = True
         return has_header
 
     def create_table(self, input_table_config: Union[dict, str]) -> None:
```

Comment:
if/else v create_table bych vytáhnul do dvou funkcí zvlášť. To by mohlo vracet nějakej rozumnej objekt. Jde jenom o zpřehlednění. Nechám na tobě.